src/GafferUI/StandardNodeGadget.cpp

```diff
@@ -69,6 +75,241 @@ using namespace Imath;
 using namespace Gaffer;
 using namespace GafferUI;
 
+namespace {
+
+static const float g_borderWidth = 0.5f;
+
+IECoreGL::Texture *focusIconTexture( bool focus, bool hover )
+{
+	static IECoreGL::TexturePtr focusIconTextures[4] = {};
+
+	if( !focusIconTextures[0] )
+	{
+		focusIconTextures[0] = ImageGadget::textureLoader()->load( "focusOff.png" );
+		focusIconTextures[1] = ImageGadget::textureLoader()->load( "focusOn.png" );
+		focusIconTextures[2] = ImageGadget::textureLoader()->load( "focusOffHover.png" );
+		focusIconTextures[3] = ImageGadget::textureLoader()->load( "focusOnHover.png" );
+
+		for( int i = 0; i < 4; i++ )
+		{
+			IECoreGL::Texture::ScopedBinding binding( *focusIconTextures[i] );
+			glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR );
+			glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR );
+			glTexParameterf( GL_TEXTURE_2D, GL_TEXTURE_LOD_BIAS, -1.0 );
+			glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_BORDER );
+			glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_BORDER );
+		}
+	}
+	return focusIconTextures[focus + 2 * hover].get();
+}
+
+//////////////////////////////////////////////////////////////////////////
+// FocusGadget
+//////////////////////////////////////////////////////////////////////////
+
+class FocusGadget : public Gadget
+{
+	public :
+
+		FocusGadget( StandardNodeGadget* parent )
+			:	m_parent( parent ),
```

Comment:
That's true - I guess I was just thinking about avoiding some polymorphic casts, but that's not really an issue.

Reply:
I guess the downside of using parent() is that then I need to wait for parenting to change, rather than doing things in the constructor?  Unlike parent(), m_parent is guaranteed to never change, since a FocusGadget is only created during a StandardNodeGadget constuctor.

Reply:
I think you can still do everything in the constructor - it's still reasonable to pass `parent` as an argument. It just seemed fragile/redundant to have an `m_parent` variable that mirrors state we already have access to via `parent()`. It might even make sense for the FocusGadget constructor to do `parent.addChild( this )` rather than do that on the outside?

Reply:
Fixed in place.